src/tools/pkcs11-tool.c

```diff
@@ -5262,12 +5309,187 @@ static int encrypt_decrypt(CK_SESSION_HANDLE session,
 		EVP_PKEY_free(pkey);
 		return 0;
 	}
-	encrypted_len = EVP_PKEY_encrypt_old(encrypted, orig_data, sizeof(orig_data), pkey);
-	EVP_PKEY_free(pkey);
-	if (((int) encrypted_len) <= 0) {
+	size_t in_len;
+	CK_ULONG mod_len = (get_private_key_length(session, privKeyObject) + 7) / 8;
+	switch (mech_type) {
+	case CKM_RSA_PKCS:
+		pad = RSA_PKCS1_PADDING;
+		/* Limit the input length to <= mod_len-11 */
```

Comment:
why?

Reply:
This limitation is coming from pkcs11-curr-v2.40.pdf
Have a look a the table p25 (2.1.6 PKCS #1 v1.5 RSA).
The Output length for C_Decrypt() must be <= k-11
(k is the length in bytes of the RSA modulus)